modules/preview-react/status-indicator/lib/StatusIndicator.tsx

```diff
@@ -13,13 +16,92 @@ export interface StatusIndicatorProps extends ExtractProps<typeof Flex, never> {
   children: React.ReactNode;
 }
 
-export const useStatusIndicator = createElemPropsHook(useStatusIndicatorModel)(({state}) => {
-  const colors = statusIndicatorColors[state.variant][state.emphasis];
-  return {
-    opacity: state.variant === 'transparent' ? '0.85' : undefined,
-    color: colors.text,
-    background: colors.background,
-  };
+const statusIndicatorStencil = createStencil({
+  base: {
+    display: 'inline-flex',
+    gap: system.space.x1,
+    maxWidth: px2rem(200),
+    alignItems: 'center',
+    borderRadius: system.shape.half,
+    height: px2rem(20),
+    padding: `${system.space.zero} ${system.space.x1}}`,
+  },
+  modifiers: {
+    gray: {
+      high: {
+        color: base.frenchVanilla100,
+        [systemIconStencil.vars.color]: base.frenchVanilla100,
+        background: base.licorice300,
+      },
+      low: {
+        color: base.licorice400,
+        [systemIconStencil.vars.color]: base.licorice400,
+        background: base.soap300,
+      },
+    },
+    orange: {
+      high: {
+        color: base.licorice500,
+        [systemIconStencil.vars.color]: base.licorice500,
+        background: base.cantaloupe400,
+      },
+      low: {
+        color: base.toastedMarshmallow600,
+        [systemIconStencil.vars.color]: base.toastedMarshmallow600,
+        background: base.cantaloupe100,
+      },
+    },
+    blue: {
+      high: {
+        color: base.frenchVanilla100,
+        [systemIconStencil.vars.color]: base.frenchVanilla100,
+        background: base.blueberry400,
+      },
+      low: {
+        color: base.blueberry500,
+        [systemIconStencil.vars.color]: base.blueberry500,
+        background: base.blueberry100,
+      },
+    },
+    green: {
+      high: {
+        color: base.frenchVanilla100,
+        [systemIconStencil.vars.color]: base.frenchVanilla100,
+        background: base.greenApple600,
+      },
+      low: {
+        color: base.greenApple600,
+        [systemIconStencil.vars.color]: base.greenApple600,
+        background: base.greenApple100,
+      },
+    },
+    red: {
+      high: {
+        color: base.frenchVanilla100,
+        [systemIconStencil.vars.color]: base.frenchVanilla100,
+        background: base.cinnamon500,
+      },
+      low: {
+        color: base.cinnamon600,
+        [systemIconStencil.vars.color]: base.cinnamon600,
+        background: base.cinnamon100,
+      },
+    },
+    transparent: {
+      high: {
+        opacity: 0.85,
+        color: base.frenchVanilla100,
+        [systemIconStencil.vars.color]: base.frenchVanilla100,
+        background: base.blackPepper600,
+      },
+      low: {
+        opacity: 0.85,
+        color: base.frenchVanilla100,
+        [systemIconStencil.vars.color]: base.frenchVanilla100,
+        background: base.blackPepper600,
+      },
+    },
+  },
```

Comment:
We will need to have default modifier too. It's handled by model now, but model will be removed in a future.

Reply:
Added back the default Modifier